### ISSUE TITLE
Colours of items in "Select feature type" menu.

### ISSUE THREAD

**Issue by eehpcm** [openstreetmap/iD#6061]
After over a year of using iD, I still haven't figured out the logic by which some items in the "Select feature type" menu get colours.  Sometimes, some of them have colours and some don't.  It's not consistent: sometimes "pond" is blue, sometimes it's not.  Sometimes "building features ..." is red, sometimes it's not.  At times I think there's a pattern, but then I find exceptions to whatever rule I dream up.

I'd find it preferable if they were always coloured instead of whatever logic is currently used.  Water features, collectively and individually, blue.  Building features and building types, red.  Natural stuff, green.  I can distinguish between colours faster than I can distinguish icons.  I know there aren't enough colours to make them all unique, but even so it would make it easier to find what I'm looking for.  Well, I'm guessing it would make life easier - I won't know for sure until/unless it happens.

Or if there is a good reason for whatever logic is currently used, could you explain it somewhere?  Because I'm baffled.

**Comment by quincylvania** [comment]
@eehpcm The fill colors are chaotic because OSM tagging is chaotic. There's no definitive categorization so icons may have received inconsistent colors as they presets were added and styled over time.

Coincidentally, the next version of iD will do a much better job of keeping the icon fill colors the same as the feature fill colors you see on the map. You can take a look on our [preview instance](http://preview.ideditor.com/master/).

I personally haven't noticed many uneven colors, but if you have any specific presets you can point out then please do!

**Comment by eehpcm** [comment]
@quincylvania 

> The fill colors are chaotic because OSM tagging is chaotic. There's no definitive categorization so icons may have received inconsistent colors as presets were added and styled over time.

Erm, you totally missed the point.  I'm not complaining about presets having colours inconsistent with other presets of the same type.  In fact, I hadn't even noticed any presets with the wrong colour compared to other presets of the same type.

I'm complaining that, at different times within the _same_ editing session, a specific preset (such as house) or preset group (such as water features) will have a colour or have no colour.  I have yet to figure out the pattern behind it.  Sometimes building features has no colour, sometimes it has a colour.  There's no rhyme or reason that I can see.

At times there's a hint of "recently-used" being a factor on whether it has colour or not, but I can see plenty of exceptions to that.  Right now, house, pond, farmyard and farm building are coloured, and those are the thee things I've used most recently and are the top of the list.  Then comes land use features, building features, water features, all monochrome.  Lake, pond, reservoir and water are next (I've only used pond and ditch recently, where's ditch in the list of recently used?) and are coloured.  Then natural features, monochrome.  Park, coloured.  Hospital grounds, monochrome (I've never used it, but it's offered to me as a choice as though I use it frequently).  Then place of worship, cafe and restaurant, coloured,  Finally, area, which I've used several times recently for multipolygons, monochrome (arguably, it probably should be monochrome).

So there's also a hint of a pattern that individual items are coloured but collections (like building features) are not.  If true, I'd say that was unhelpful (if I'm looking for pond and the only thing in the list is water features, having that blue would help me spot it).  But it's not true.  Because, for unknown reasons, I've also seen building features as red as house, water features as blue as pond, etc.

BTW, a suitable colour for land use features would be rainbow. :)

**Comment by quincylvania** [comment]
@eehpcm Oh I believe you mean that the icons for the expandable categories don't have the colors you expect?

<img width="376" alt="Screen Shot 2019-03-20 at 10 06 16 AM" src="https://user-images.githubusercontent.com/2046746/54690497-ecb25b00-4af7-11e9-9469-18c33c524f92.png">

I think a solution would be to avoid putting a fill on category icons at all so they're less confusable with actual presets. It actually works like this in the new feature-adding workflow in the next release of iD:

<img width="512" alt="Screen Shot 2019-03-20 at 10 08 05 AM" src="https://user-images.githubusercontent.com/2046746/54690726-64808580-4af8-11e9-88c5-cbaa987845e5.png">

If you do see the exact same presets have different colors at different times, please take screenshots if you can and let us know.

**Comment by eehpcm** [comment]
@quincylvania 

> Oh I believe you mean that the icons for the expandable categories don't have the colors you expect?

That's part of it.  Because, occasionally, the expandable categories _do_ have the colours I expect.  Unless I imagined that.  And occasionally, the actual presets do _not_ have a colour when at other times the _same_ preset does have a colour.  I notice it with ponds, which are usually blue but sometimes (rarely) not.

> I think a solution would be to avoid putting a fill on category icons at all so they're less confusable with actual presets. 

Noooooooooooooooooooooooo!

That list of presets and expandables is, as far as I can tell, random.  Things appear in different orders at different times after I've drawn an area.  There's an element of recently-used to it, but the rest of it might as well be random because I can't predict what will be on the list or where it will appear in the list.  Colour _everything_.  So that after I've drawn an area for a pond, I can scan for _blue._  That way I might see the pond preset if I've recently used it, or I might see water features, and either will do.  If expandables are monochrome then if I don't see blue pond I have to examine icons to find water features, and icons are slower to spot than colours are.  Colour recognition is faster than pattern recognition.

Maybe you could invert the areas that are coloured for expandables or something like that.  Black drop on blue for pond, blue drop on grey for water features.  Something that lets me use colour as a quick way of finding what I'm looking for, so that I only have to examine icons of a specific colour.

**Comment by bhousel** [comment]
@eehpcm I can tell this is an issue that you care a lot about.

Try to understand... @quincylvania and I have no idea what you're talking about.  We _think_ that we already do color the things in iD in a consistent way.  Water is blue, trees are green, buildings are red, sand is yellow, parking lots are dark grey, etc etc.  Miscellaneous things default to grey if they don't have a better color assigned.

Nobody has ever told us that iD is showing random colors before, so it really seems to us like this issue is coming out of nowhere.  

I'd like you to feel better about how iD works, but at the same time, we really don't understand your issue with the colors.  You keep using the word "random" but the colors are definitely not that.  Please take a screenshot of something that you find confusing and let us know what we're doing wrong, because we don't see it..

**Comment by eehpcm** [comment]
@bhousel 

> I can tell this is an issue that you care a lot about.

The issue itself, not so much.  It's a minor problem.  Trying to communicate exactly what the issue is, however, seems to be more taxing.

The colours _when they show up_ are not random.  Whether or not the colours show up is random.  Most times the pond preset in recently-used is blue.  Rarely, and for no discernible reason, it's monochrome.  Same for any other preset.  Usually, the expandables are monochrome.  Rarely, and for no discernible reason, they have the colour I'd expect (if they were meant to have a colour, which apparently they aren't).

It's happened for as long as I've used iD (around a year).  It's rare, though.

I may have been imagining the expandables occasionally being coloured because that's how I'd like it to work.

Next time I notice it I'll screenshot it.

**Comment by eehpcm** [comment]
OK, I'm not imagining it.  And I have a [screenshot](https://user-images.githubusercontent.com/3224921/54705361-88fa5300-4b34-11e9-8d38-dfb3f2802d3b.png) to show it.  I just mapped a stream.  In select features I was offered a blue ditch (I'd recently mapped a ditch) and also water features.  Water features was, for some inexplicable reason, blue (normally it isn't, it's only rarely blue).  I'd like it to be blue _every_ time.  Apparently Quincy thinks that's a bad thing, but I like it on the rare occasions it happens, because it's a lot easier to spot  EIther way it ought to be consistent.

**Comment by bhousel** [comment]
Ok, yeah... As @quincylvania said above, we don't add colors to the categories.  This is really because they can contain all kinds of things.  I agree with his suggestion to style them somehow differently to make them look less similar to presets.

**Comment by quincylvania** [comment]
@eehpcm So the behavior in your screenshot isn't inexplicable. It's because there are two different "Water" categories, one for lines and one for areas. Their icons are different. We're changing this somewhat in the next release of iD where presets of different geometry are mixed together in the search results.

**Comment by eehpcm** [comment]
@quincylvania 

> So the behavior in your screenshot isn't inexplicable. It's because there are two different "Water" categories, one for lines and one for areas. Their icons are different

OK, so now I can predict it.  Until you explained, I wasn't sure what had triggered it in this case.  Because I'd mapped a line?  Because immediately prior to adding the stream I'd added a pond?  But now you've explained it I am no longer puzzled.

>  We're changing this somewhat in the next release of iD where presets of different geometry are mixed together in the search results.

Please can we have coloured expandable categories?  I don't know about others, but it makes life a lot easier for me.  If I've just drawn a watery thing, it's a lot faster to look for blue in the select features.  If the exact one I want isn't there then I can home in on water features quickly.  By all means emphasise the difference between expandable and preset but please colour-code the expandables (at least where the bulk of their presets share the same colour).

**Comment by bhousel** [comment]
> Please can we have coloured expandable categories? I don't know about others, but it makes life a lot easier for me. If I've just drawn a watery thing, it's a lot faster to look for blue in the select features. If the exact one I want isn't there then I can home in on water features quickly. By all means emphasise the difference between expandable and preset but please colour-code the expandables (at least where the bulk of their presets share the same colour).

No, we can not have colored categories.  Too many of them are like "golf" or "natural" or "landuse" and can have a bunch of things inside.

The feature you just mapped is automatically put at the top of the preset list, and it has the color it should have, so your eyes should be able to find it quickly.  You don't need to use the categories in this case.  We'll try to make the categories look different, but not by making them look like presets.

**Comment by eehpcm** [comment]
@bhousel 

> No, we can not have colored categories. Too many of them are like "golf" or "natural" or "landuse" and can have a bunch of things inside.

And some of them, like water features, have blue presets inside.  Nothing but blue presets.  The same is almost true of building features.

> The feature you just mapped is automatically put at the top of the preset list, and it has the color it should have, so your eyes should be able to find it quickly. 

That only works if I've just mapped the same type of feature I did before.  Sometimes that's true, sometimes it's not.  I just mapped a farm house, farm buildings, a farmyard, a pond and a stream.  Of those, only the farm buildings were repetitive.  The screenshot was from that session.  The stream was not in that list of presets because it had been a while since I'd added a stream.  The fact that water features was blue let me spot it quickly.

I'll give up here.  If I've not persuaded you by now I doubt I ever will.
